### DISCUSSION TITLE
kafka routine load如何支持消息中的换行符

### DISCUSSION THREAD

**Discussion by cdavis26424** [apache/doris#6798] [A - General / Q&A] [answered]
kafka版本：2.4.1
Doris version 0.12.0-rc03

由于数据源某字符串类型字段中包含换行，导致在kafka消息中也有换行，这些带有换行的消息都没有导入到doris。
尝试了将消息中的换行都替换成\r\n，结果doris中存储的也是\r\n。
测试了使用sql insert带有\r\n的数据，发现在doris中可以转义为换行符。

job构建语句
CREATE ROUTINE LOAD my_db.my_job ON my_table
COLUMNS TERMINATED BY "|",
COLUMNS(cola, colb, colc)
PROPERTIES
(
    "desired_concurrent_number"="3",
    "max_batch_interval" = "20",
    "max_batch_rows" = "300000",
    "max_batch_size" = "209715200",
    "max_error_number" = "30000",
    "strict_mode" = "false"
)
FROM KAFKA
(
    "kafka_broker_list" = "192.168.0.131:9092,192.168.0.132:9092,192.168.0.133:9092",
    "kafka_topic" = "my_topic",
    "property.group.id" = "my_consumers"
);

**Comment by morningman** [comment, marked as answer]
版本太老了，建议升级。新版本支持json格式，或者通过多字节分隔符解决这个问题。
